pkg/idp/third_party.go

```diff
@@ -55,14 +55,14 @@ type Configuration struct {
 	// AuthURL is the authorize url, typically something like: https://example.org/oauth2/auth
 	// Should only be used when the OAuth2 / OpenID Connect server is not supporting OpenID Connect Discovery and when
 	// `provider` is set to `generic`.
-	// validate that this field is required only when Provider field == "generic"
-	AuthURL string `json:"auth_url" yaml:"auth_url" validate:"required_if=Provider generic"`
+	// validate that this field is required only when Provider field == "generic" and IssuerURL is empty
+	AuthURL string `json:"auth_url" yaml:"auth_url"`
```

Comment:
IMHO we don't need the custom validator. This seems to work:
```
AuthURL string `json:"auth_url" yaml:"auth_url" validate:"excluded_with=IssuerURL&&required_if=Provider generic"`
```

Reply:
The `excluded_with` doesn't appear in the linked sections of the doc like the other tags, so I did not see that.
I don't see documentation for it. Also, the && is a comma or I should test the actual ampersand?

Reply:
I tested your solution, I don't like the use of the ampersand because it is not supported by the library.
The `and` is expressed through the comma `,`, and with that it doesn't work.
Based on these findings, I have to stick with the custom function (as much as I don't fancy it).